src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,8 +1,43 @@
 package core.basesyntax;
 
+import java.lang.reflect.Array;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private static final DateTimeFormatter DATE_FORMAT = DateTimeFormatter.ofPattern("dd.MM.yyyy");
 
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder builder = new StringBuilder();
+        builder.append("Report for period ")
+                .append(dateFrom)
+                .append(" - ")
+                .append(dateTo)
+                .append("\n");
+        for (String name: names) {
+            int salary = 0;
+            for (String piece: data) {
+                if (LocalDate.parse(Array.get(piece.split(" ", 0), 0).toString(), DATE_FORMAT)
+                        .compareTo(LocalDate.parse(dateFrom, DATE_FORMAT)) >= 0 
+                        && LocalDate.parse(Array.get(piece.split(" ", 0), 0)
+                                .toString(), DATE_FORMAT)
+                            .compareTo(LocalDate.parse(dateTo, DATE_FORMAT)) <= 0 
+                        && Array.get(piece.split(" ", 0), 1).toString().equals(name)) {
+                    salary += getSalary(
+                            Integer.parseInt(Array.get(piece.split(" ", 0), 2).toString()),
+                            Integer.parseInt(Array.get(piece.split(" ", 0), 3).toString()));
```

Comment:
you execute split 5 times at least that will generate a lot of unneeded objects, use it once and save result into variable